src/services/auth/handlers/postAuth.ts

```diff
@@ -0,0 +1,77 @@
+import { Handler } from "aws-lambda";
+import * as cognitolib from "../../../libs/cognito-lib";
+
+if (!process.env.apiKey) {
+  throw "ERROR:  process.env.apiKey is required,";
+}
+const apiKey: string = process.env.apiKey;
+
+if (!process.env.apiEndpoint) {
+  throw "ERROR:  process.env.apiEndpoint is required,";
+}
+const apiEndpoint: string = process.env.apiEndpoint;
+
+export const handler: Handler = async (event, context) => {
+  console.log(JSON.stringify(event,null,2));
+  const { request, response } = event;
+  const { userAttributes } = request;
+
+  if(!userAttributes.identities){
+    console.log("User is not managed externally.  Nothing to do.");
+  } else {
+    console.log("Getting user attributes from external API");
+  
+    try {
+      const username = userAttributes["custom:username"]; // This is the four letter IDM username
+      const response = await fetch(`${apiEndpoint}/api/v1/authz/id/all?userId=${username}`, {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+          'x-api-key': apiKey,
+        }
+      });
+      if (!response.ok) {
+        console.log(response)
+        throw new Error('Network response was not ok');
```

Comment:
Does the `response` come back with anything useful that we'd wanna throw in the `Error` to help with error logging??